### PR TITLE
Add support for token based access to filesystem

### PR DESCRIPTION
**Pull request by vermapratyush**
Some filesystem have token based authorization model. Add refreshAccessToken method to allow filesystem to implement their token fetch mechanism.

### CODE REVIEW

File: velox/common/file/FileSystems.cpp [hunk 1]
```diff
@@ -61,6 +61,16 @@ std::shared_ptr<FileSystem> getFileSystem(
   VELOX_FAIL("No registered file system matched with file path '{}'", filePath);
 }
 
+std::vector<std::shared_ptr<filesystems::FileSystem>>
+getRegisteredFileSystems() {
```
Comment (mbasmanova) [inline velox/common/file/FileSystems.cpp:65]: Unrelated change?

Comment (vermapratyush) [reply velox/common/file/FileSystems.cpp:65]: This is used here: https://github.com/prestodb/presto/pull/20457/files#diff-baa23cbd705092e9a2cc187cc5ca1237bd72734427b5ef7afb4e94ace27cd2e7R331-R336
